addons/udes_stock/models/stock_picking_batch.py

```diff
@@ -298,63 +303,128 @@ def get_next_task(self, skipped_product_ids=None, task_grouping_criteria=None):
         They are enabled by picking type and should be filled at
         _prepare_task_info(), by default it is not required to confirm anything.
         """
+        MoveLine = self.env['stock.move.line']
+
         self.ensure_one()
 
-        available_mls = self.get_available_move_lines()
+        all_available_mls = self.get_available_move_lines()
+        skipped_mls = MoveLine.browse()
 
-        if skipped_product_ids is not None:
-            available_mls = available_mls.filtered(
-                lambda ml: ml.product_id.id not in skipped_product_ids
-            )
+        # Filter out skipped move lines
+        if skipped_product_ids:
+            skipped_mls = all_available_mls.filtered(
+                lambda ml: ml.product_id.id in skipped_product_ids)
+        elif skipped_move_line_ids:
```

Comment:
We only care about skipped_move lines if skipped_product_ids is Falsey?

Reply:
Yeah they should be mutually exclusive. I'll add a check for this at the beginning of the function